src/renderer/library/components/settings/Settings.tsx

```diff
@@ -36,10 +43,48 @@ class Settings extends React.Component<IProps, undefined> {
                 <LibraryLayout
                     title={__("header.settings")}
                 >
```

Comment:
This component is loaded from react-router https://github.com/edrlab/thorium-reader/blob/1f0decaa938b5cdf621da84804fdda193aed6bac/src/renderer/library/routing.ts#L62-L66

I think that the react-router route `/setting` should be disable (the setting component is no longer a page but a modal dialog now). And Setting Dialog must be move to the `common/components/dialog` folder (used in library and reader as well)

Reply:
> I think that the react-router route `/setting` should be disable (the setting component is no longer a page but a modal dialog now).

Nested routes (i.e. relative `settings_modal` path inside another route definition) would work well declaratively, for example `/bookshelf` + nested `/bookshelf/settings_modal` ... but the settings modal popup dialog with Radix is really designed to be displayed anywhere (overlaid on top of an existing route component) where a "call site" button (trigger UI control) can be placed, so I agree that using the router to implement this is a waste of time and effort (unless we really want to use router functionality, for example to open the settings dialog in a specific tab by passing route parameters?)